src/plugins/orangehrmPerformancePlugin/Api/ReviewKpiAPI.php

```diff
@@ -0,0 +1,121 @@
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Performance\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\PerformanceReview;
+use OrangeHRM\Performance\Api\Model\KpiModel;
+use OrangeHRM\Performance\Dto\ReviewKpiSearchFilterParams;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Performance\Traits\Service\PerformanceReviewServiceTrait;
+
+class ReviewKpiAPI extends Endpoint implements CollectionEndpoint
+{
+    use PerformanceReviewServiceTrait;
+    use UserRoleManagerTrait;
+
+    public const PARAMETER_REVIEW_ID = 'reviewId';
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $reviewKpiParamHolder = new ReviewKpiSearchFilterParams();
+        $reviewKpiParamHolder->setReviewId(
+            $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_ATTRIBUTE,
+                self::PARAMETER_REVIEW_ID
+            )
+        );
+        $accessibleIds = $this->getUserRoleManager()->getAccessibleEntityIds(PerformanceReview::class);
+        if (! in_array($reviewKpiParamHolder->getReviewId(), $accessibleIds)) {
+            throw $this->getBadRequestException();
+        }
+        $this->setSortingAndPaginationParams($reviewKpiParamHolder);
+        $reviewKpis = $this->getPerformanceReviewService()->getPerformanceReviewDao()
+            ->getKpisForReview($reviewKpiParamHolder);
+        $reviewCount = $this->getPerformanceReviewService()->getPerformanceReviewDao()
+            ->getKpisForReviewCount($reviewKpiParamHolder);
+
+        return new EndpointCollectionResult(
+            KpiModel::class,
+            $reviewKpis,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL=>$reviewCount])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_REVIEW_ID,
+                new Rule(Rules::POSITIVE)
```

Comment:
@devishke-orange seems we can move permission checking to here

Reply:
Changed to here and with testing it works